Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -78,42 +79,70 @@ public function apply(ResourceInterface $resource, QueryBuilder $queryBuilder)
             return;
         }
 
-        $metadata = $this->getClassMetadata($resource);
-        $fieldNames = array_flip($metadata->getFieldNames());
-
         foreach ($this->extractProperties($request) as $property => $value) {
-            if (null === $value || !$this->isPropertyEnabled($property)) {
+            if (
+                !$this->isPropertyEnabled($property)
+                || !$this->isPropertyMapped($property, $resource, true)
+                || null === $value
+            ) {
                 continue;
             }
 
-            if (isset($fieldNames[$property])) {
+            $alias = 'o';
+            $field = $property;
+
+            if ($this->isPropertyNested($property)) {
+                $propertyParts = $this->splitPropertyParts($property);
+
+                $parentAlias = $alias;
+
+                foreach ($propertyParts['associations'] as $association) {
+                    $alias = QueryUtils::generateJoinAlias($association);
+                    $queryBuilder->join(sprintf('%s.%s', $parentAlias, $association), $alias);
```

Comment:
Can you be more specific? Because I haven't been able to reproduce it... I've tried filtering two properties belonging to the same nested resource at the same time and it works just fine.